### PR TITLE
Synchronous/asynchronous Promise rejection

### LINKED ISSUE #14: cropTo is returning promises but it is unclear when promises are supposed to be rejected synchronously or not
For promise-based operations, we usually use language like:
1. check if state1 is ok, if not return a rejected promise // synchronously rejected promise.
2. check if state2 is ok, if not return a rejected promise // synchronously rejected promise.
3. Create a promise
4. Run the following steps in parallel: //asynchronously settled promise.
5. return promise.

It seems like the self-capture track check step can be made synchronously.
It seems the MUST validate cropTarget step should be done asynchronously, as part of a 'run steps in parallel' section.

Also the current algorithm seems to check for valid crop target twice (step 2 and step 4.2). Probably step 4.2 valid CropTarget is unneeded? Or could it be that a crop target is valid in step 2 but invalid in step 4.2.

### PR DESCRIPTION
**Pull request by eladalon1983**
Fixes #14.

***
<a href="https://pr-preview.s3.amazonaws.com/eladalon1983/mediacapture-region/pull/28.html" title="Last updated on Mar 24, 2022, 4:21 PM UTC (ccea3b0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-region/28/81c25e0...eladalon1983:ccea3b0.html" title="Last updated on Mar 24, 2022, 4:21 PM UTC (ccea3b0)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -259,43 +262,50 @@ <h3>BrowserCaptureMediaStreamTrack</h3>
...
-                  any additional frames delivered to the application will therefore be cropped (or
-                  uncropped) according to either <var>POST-STATE</var> or a later state.
-                </p>
-              </li>
+              <li>Let <var>P</var> be a new {{Promise}}.</li>
+              <li>Run the following steps in parallel:</li>
+              <ol>
+                <li>
+                  If <var>cropTarget</var> is neither {{undefined}} nor a [=valid CropTarget=],
+                  reject <var>P</var> with a {{NotAllowedError}} and abort these steps.
```
[the remark this answers is not in the document]
Comment (youennf) [reply index.html:274]: I filed https://github.com/w3c/mediacapture-region/issues/29

File: index.html [hunk 2]
```diff
@@ -259,43 +262,50 @@ <h3>BrowserCaptureMediaStreamTrack</h3>
...
-                <p>
-                  Call the track's state before this method invocation <var>PRE-STATE</var>, and
-                  after this method invocation <var>POST-STATE</var>. The user agent MUST resolve
-                  <var>P</var> when it is guaranteed that no more frames cropped (or uncropped)
-                  according to <var>PRE-STATE</var> have been delivered to the application, and that
-                  any additional frames delivered to the application will therefore be cropped (or
-                  uncropped) according to either <var>POST-STATE</var> or a later state.
-                </p>
-              </li>
+              <li>Let <var>P</var> be a new {{Promise}}.</li>
```
[the remark this answers is not in the document]
Comment (eladalon1983) [reply index.html:269]: Done.

File: index.html [hunk 3]
```diff
@@ -259,43 +262,50 @@ <h3>BrowserCaptureMediaStreamTrack</h3>
                   </li>
                 </ul>
                 <p>
-                  When the user agent does not accept <var>cropTarget</var>, it MUST return a new
+                  If the user agent does not accept <var>cropTarget</var>, it MUST return a new
```
[the remark this answers is not in the document]
Comment (eladalon1983) [reply index.html:265]: Done.

File: index.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (eladalon1983) [reply index.html]: > This might be best handled as a separate issue.

Agreed.

**Review by jan-ivar — APPROVED** [review]
LGTM for now since we have #29 to clean this up

**Pull request merged**